src/parser/mod.rs

```diff
@@ -4529,7 +4531,26 @@ impl<'a> Parser<'a> {
         })
     }
 
-    fn parse_drop_function_desc(&mut self) -> Result<DropFunctionDesc, ParserError> {
+    /// ```sql
+    /// DROP PROCEDURE [ IF EXISTS ] name [ ( [ [ argmode ] [ argname ] argtype [, ...] ] ) ] [, ...]
+    /// [ CASCADE | RESTRICT ]
+    /// ```
+    fn parse_drop_procedure(&mut self) -> Result<Statement, ParserError> {
+        let if_exists = self.parse_keywords(&[Keyword::IF, Keyword::EXISTS]);
+        let proc_desc = self.parse_comma_separated(Parser::parse_drop_function_desc)?;
+        let option = match self.parse_one_of_keywords(&[Keyword::CASCADE, Keyword::RESTRICT]) {
+            Some(Keyword::CASCADE) => Some(ReferentialAction::Cascade),
+            Some(Keyword::RESTRICT) => Some(ReferentialAction::Restrict),
+            _ => None,
```

Comment:
```suggestion
            Some(invalid) => return parser_err!(...)
            None => None,
```
Should this rather return an error if the action is unexpected?
seems otherwise e.g. `DROP PROCEDURE testproc SET NULL` will be accepted by the parser but it will silently discard the `SET NULL` part (we can also include a test case demonstrating the behavior)

Reply:
I think they way that parse_one_of_keywords works, it will only return CASCADE or RESTRICT here (never Some(..)). 

I added a test in e782bf5 and changed the code to try and make this clearer